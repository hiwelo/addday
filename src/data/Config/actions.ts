import { ActionsUnion, ConfigState } from '../types';

/**
 * Resets the current user for this application
 */
function setSentryEnrollment(
  updatedEnrollmentStatus: ConfigState['sentryEnrollment'],
): ActionsUnion {
  return {
    type: 'SENTRY_ENROLLMENT',
    payload: updatedEnrollmentStatus,
  };
}

export { setSentryEnrollment };
