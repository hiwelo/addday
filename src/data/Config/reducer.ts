import { ConfigState, ConfigAction } from './types';

const INITIAL_STATE: ConfigState = {
  sentryEnrollment: false,
};

function configReducer(
  state = INITIAL_STATE,
  action: ConfigAction,
): ConfigState {
  const { type, value } = action;

  switch (type) {
    case 'SENTRY_ENROLLMENT':
      return {
        ...state,
        sentryEnrollment: value.sentryEnrollment,
      };

    default:
      return state;
  }
}

export default configReducer;
