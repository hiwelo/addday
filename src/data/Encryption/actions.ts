import { ActionsUnion } from '../types';

/**
 * Fetches the encryption key
 */
function getEncryptionKey(): ActionsUnion {
  return {
    type: 'GET_ENCRYPTION_KEY',
  };
}

/**
 * Saves a new encryption key
 */
function setEncryptionKey(key: string): ActionsUnion {
  return {
    type: 'SET_ENCRYPTION_KEY',
    payload: key,
  };
}

export { getEncryptionKey, setEncryptionKey };
