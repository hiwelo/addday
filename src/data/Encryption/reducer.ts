import { ActionsUnion, EncryptionState } from '../types';

function encryptionReducer(
  state: EncryptionState = { key: null },
  { type }: ActionsUnion,
): EncryptionState {
  switch (type) {
    case 'GET_ENCRYPTION_KEY':
      return state;

    case 'SET_ENCRYPTION_KEY':
      return {
        ...state,
        key: 'test',
      };

    default:
      return state;
  }
}

export default encryptionReducer;
