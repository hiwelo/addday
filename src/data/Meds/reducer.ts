import { ActionsUnion, MedsState } from '../types';

const INITIAL_STATE: MedsState = {
  intakes: new Map(),
  medications: {},
};

function medsReducer(
  state: MedsState = INITIAL_STATE,
  action: ActionsUnion,
): MedsState {
  switch (action.type) {
    case `CLEAR_SCHEDULED_INTAKES_ACTION`: {
      state.intakes.clear();

      return state;
    }

    case `CLEAR_SCHEDULED_MEDICATION_INTAKES_ACTION`: {
      const scheduledIntake = state.intakes.get(action.payload);

      if (!scheduledIntake) {
        throw new Error('The requested scheduled intake was not found.');
      }

      scheduledIntake.medications.clear();

      return {
        ...state,
        intakes: state.intakes.set(scheduledIntake.id, {
          ...scheduledIntake,
        }),
      };
    }

    case `DELETE_SCHEDULED_INTAKE_ACTION`: {
      state.intakes.delete(action.payload);

      return state;
    }

    case `DELETE_SCHEDULED_MEDICATION_INTAKE_ACTION`: {
      const scheduledIntake = state.intakes.get(
        action.payload.scheduledIntakeId,
      );

      if (!scheduledIntake) {
        throw new Error('The requested scheduled intake was not found.');
      }

      scheduledIntake.medications.delete(action.payload.scheduledMedicationId);

      return {
        ...state,
        intakes: state.intakes.set(scheduledIntake.id, {
          ...scheduledIntake,
        }),
      };
    }

    case `RESET_APP`:
      return INITIAL_STATE;

    case `SET_SCHEDULED_INTAKE_ACTION`: {
      state.intakes.set(action.payload.id, action.payload);

      return { ...state };
    }

    case `SET_SCHEDULED_MEDICATION_INTAKE_ACTION`: {
      const scheduledIntake = state.intakes.get(
        action.payload.scheduledIntakeId,
      );

      if (!scheduledIntake) {
        throw new Error('The requested scheduled intake was not found.');
      }

      const providedScheduledMedicationIntake =
        action.payload.scheduledMedicationIntake;
      const updatedScheduledMedications = scheduledIntake.medications.set(
        providedScheduledMedicationIntake.medicationId,
        providedScheduledMedicationIntake,
      );

      return {
        ...state,
        intakes: state.intakes.set(scheduledIntake.id, {
          ...scheduledIntake,
          medications: updatedScheduledMedications,
        }),
      };
    }

    case 'UPDATE_MEDICATION':
      return {
        ...state,
        medications: {
          ...state.medications,
          [action.payload.id]: action.payload,
        },
      };

    default:
      return state;
  }
}

export default medsReducer;
