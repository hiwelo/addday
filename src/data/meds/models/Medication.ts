export interface Medication {
  /** Medication unique ID */
  id: string;
  /** Medication name */
  name: string;
  /** Medication's name alias to use in notifications */
  alias?: string;
}
