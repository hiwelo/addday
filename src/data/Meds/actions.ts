import { Medication } from '../../models/Medication';
import { ActionsUnion } from '../types';

export function editMedication(medication: Medication): ActionsUnion {
  return {
    type: 'UPDATE_MEDICATION',
    payload: medication,
  };
}
