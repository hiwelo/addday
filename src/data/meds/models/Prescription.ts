import { Medication } from './Medication';

/** Days to use for the intake definition */
type IntakeDay =
  | 'Monday'
  | 'Tuesday'
  | 'Wednesday'
  | 'Thursday'
  | 'Friday'
  | 'Saturday'
  | 'Sunday';

/** Kind of intake supported */
type IntakeMode = 'interval' | 'on-demand' | 'scheduled';

export type IntakeWindow = {
  /** Days when the intake is possible */
  days: IntakeDay[];
  /** Intake timeframe end point */
  end: string;
  /** Intake timeframe start point */
  start: string;
};

export interface Prescription {
  /** Intake mode for this prescription */
  intakeMode: IntakeMode;
  /** Medication prescribed */
  medication: Medication;
  /** Windows when the medication is available */
  availability?: IntakeWindow[];
  /** Maximum number of intakes per day before displaying a warning */
  maxDailyIntake?: number;
  /** Medication strength */
  strength?: string;
}

export interface IntervalBasedPrescription extends Prescription {
  intakeMode: 'interval';
}

export interface OnDemandPrescription extends Prescription {
  intakeMode: 'on-demand';
}

export interface ScheduledPrescription extends Prescription {
  availability: undefined;
  intakeMode: 'scheduled';
  strength: undefined;
}
