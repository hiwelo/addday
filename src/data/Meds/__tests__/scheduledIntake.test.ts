import moment from 'moment';

import { WeeklySchedule } from '../../../models/WeeklySchedule';
import * as actions from '../actions';
import reducer from '../reducer';

const mockedMoment = moment.utc('18:00', 'HH:mm');

const mockedWeeklySchedule: WeeklySchedule = {
  monday: true,
  tuesday: false,
  wednesday: false,
  thursday: true,
  friday: true,
  saturday: false,
  sunday: false,
};

const testScheduledIntake = {
  days: mockedWeeklySchedule,
  id: 'test',
  moment: mockedMoment,
  medications: new Map(),
};

describe('Scheduled Intakes', () => {
  it('allows you to add a new scheduled intake', () => {
    const action = actions.setScheduledIntake(testScheduledIntake);
    const state = reducer(undefined, action);

    expect(action).toMatchSnapshot();
    expect(state).toMatchSnapshot();
  });

  it('allows you to update an existing scheduled intake', () => {
    const action = actions.setScheduledIntake(testScheduledIntake);
    const state = reducer(undefined, action);
    const updateAction = actions.setScheduledIntake({
      ...testScheduledIntake,
      medications: new Map().set('test', {
        medicationId: 'test',
        units: 2,
      }),
    });
    const updatedState = reducer(state, updateAction);

    expect(updateAction).toMatchSnapshot();
    expect(updatedState).toMatchSnapshot();
  });

  it('allows you to delete an existing scheduled intake', () => {
    const action = actions.setScheduledIntake(testScheduledIntake);
    const state = reducer(undefined, action);
    const updateAction = actions.deleteScheduledIntake('test');
    const updatedState = reducer(state, updateAction);

    expect(updateAction).toMatchSnapshot();
    expect(updatedState).toMatchSnapshot();
  });

  it('allows you to clear all existing scheduled intake', () => {
    const action1 = actions.setScheduledIntake(testScheduledIntake);
    const action2 = actions.setScheduledIntake({
      ...testScheduledIntake,
      id: 'test2',
    });
    const state = reducer(undefined, action1);
    const updatedState = reducer(state, action2);
    const clearAction = actions.clearScheduledIntakes();
    const clearedState = reducer(updatedState, clearAction);

    expect(clearAction).toMatchSnapshot();
    expect(clearedState).toMatchSnapshot();
  });
});
