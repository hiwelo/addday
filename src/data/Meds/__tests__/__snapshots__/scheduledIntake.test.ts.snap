// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scheduled Intakes allows you to add a new scheduled intake 1`] = `
Object {
  "payload": Object {
    "days": Map {
      "monday" => true,
      "tuesday" => false,
      "wednesday" => false,
      "thursday" => true,
      "friday" => true,
      "saturday" => false,
      "sundary" => false,
    },
    "id": "test",
    "medications": Map {},
    "moment": "2020-05-03T18:00:00.000Z",
  },
  "type": "SET_SCHEDULED_INTAKE_ACTION",
}
`;

exports[`Scheduled Intakes allows you to add a new scheduled intake 2`] = `
Object {
  "intakes": Map {
    "test" => Object {
      "days": Map {
        "monday" => true,
        "tuesday" => false,
        "wednesday" => false,
        "thursday" => true,
        "friday" => true,
        "saturday" => false,
        "sundary" => false,
      },
      "id": "test",
      "medications": Map {},
      "moment": "2020-05-03T18:00:00.000Z",
    },
  },
  "medications": Object {},
}
`;

exports[`Scheduled Intakes allows you to clear all existing scheduled intake 1`] = `
Object {
  "type": "CLEAR_SCHEDULED_INTAKES_ACTION",
}
`;

exports[`Scheduled Intakes allows you to clear all existing scheduled intake 2`] = `
Object {
  "intakes": Map {},
  "medications": Object {},
}
`;

exports[`Scheduled Intakes allows you to delete an existing scheduled intake 1`] = `
Object {
  "payload": "test",
  "type": "DELETE_SCHEDULED_INTAKE_ACTION",
}
`;

exports[`Scheduled Intakes allows you to delete an existing scheduled intake 2`] = `
Object {
  "intakes": Map {},
  "medications": Object {},
}
`;

exports[`Scheduled Intakes allows you to update an existing scheduled intake 1`] = `
Object {
  "payload": Object {
    "days": Map {
      "monday" => true,
      "tuesday" => false,
      "wednesday" => false,
      "thursday" => true,
      "friday" => true,
      "saturday" => false,
      "sundary" => false,
    },
    "id": "test",
    "medications": Map {
      "test" => Object {
        "medicationId": "test",
        "units": 2,
      },
    },
    "moment": "2020-05-03T18:00:00.000Z",
  },
  "type": "SET_SCHEDULED_INTAKE_ACTION",
}
`;

exports[`Scheduled Intakes allows you to update an existing scheduled intake 2`] = `
Object {
  "intakes": Map {
    "test" => Object {
      "days": Map {
        "monday" => true,
        "tuesday" => false,
        "wednesday" => false,
        "thursday" => true,
        "friday" => true,
        "saturday" => false,
        "sundary" => false,
      },
      "id": "test",
      "medications": Map {
        "test" => Object {
          "medicationId": "test",
          "units": 2,
        },
      },
      "moment": "2020-05-03T18:00:00.000Z",
    },
  },
  "medications": Object {},
}
`;
